reservation_units/utils/reservation_unit_reservation_scheduler.py

```diff
@@ -136,8 +136,11 @@ def _get_reservation_period_start(self) -> datetime.datetime:
 
         return DEFAULT_TIMEZONE.localize(datetime.datetime.now() + delta)
 
-    def _get_reservation_period_end(self, start: datetime.date) -> datetime.date:
-        if self.reservation_unit.reservation_ends:
+    def _get_reservation_period_end(self, start: datetime.datetime) -> datetime.date:
+        if (
+            self.reservation_unit.reservation_ends
+            and self.reservation_unit.reservation_ends > start
+        ):
             return self.reservation_unit.reservation_ends.date()
```

Comment:
Again not part of this PR, and I'm not actually sure about this, but I think this might give the date based on UTC timezone, so it will be incorrect when the time is close to midnight. That is probably not very common in practice though 😄